recipes-tests/trenchboot-tests/trenchboot-tests/cukinia.conf

```diff
@@ -0,0 +1,74 @@
+export CUKINIA_ROOT=/etc/cukinia
+
+shopt -s nullglob
+
+check_acm() {
+    files=("/boot/acm/$1"*.bin)
+    if [ ${#files[@]} -eq 0 ]; then
+        return 1
+    fi
+    for acm in "${files[@]}"; do
+        if [ -f "$acm" ]; then
+            grep "/acm/$(basename "$1")" /boot/grub/grub.cfg || return 1
+        else
+            return 1
+        fi
+    done
+}
+
+acms=(
+    ADL_SINIT
+    BDW_SINIT
+    CFL_SINIT
+    CMLSTGP_SINIT
+    CML_RKL_S_SINIT
+    CML_S_SINIT
+    RKLS_SINIT
+    SKL_KBL_AML_SINIT
+    SNB_IVB_SINIT
+    TGL_SINIT
+)
+
+cukinia_log "$(_colorize yellow "--- GRUB ---")"
+cukinia_test -f /boot/grub/grub.cfg
+cukinia_test -f /boot/EFI/BOOT/grub.cfg
+as "Check if /boot/grub/grub.cfg and /boot/EFI/BOOT/grub.cfg are the same" \
+    cukinia_cmd cmp /boot/grub/grub.cfg /boot/EFI/BOOT/grub.cfg
+as "grub.cfg contains slaunch" cukinia_cmd grep 'slaunch\s*$' /boot/grub/grub.cfg
+
+cukinia_log "$(_colorize yellow "--- ACM & SKL ---")"
+for acm in "${acms[@]}"; do
+    as "/boot/acm/$acm*.bin exists and is used in grub.cfg" cukinia_cmd check_acm "$acm"
+done
+cukinia_test -f /boot/skl.bin
+as "skl.bin is used in grub.cfg" cukinia_cmd grep "/skl.bin" /boot/grub/grub.cfg
+
+cukinia_log "$(_colorize yellow "--- packages ---")"
+as "Check whether the tpm2_tools package can be started" cukinia_cmd tpm2_eventlog -v
+as "Check whether the rsync package can be started" cukinia_cmd rsync --version
+
+cukinia_log "$(_colorize yellow "--- TPM ---")"
+# cukinia_test -f /sys/kernel/security/tpm0/binary_bios_measurements
+# as "TPM event log can be read by tpm2_eventlog" \
+#     cukinia_cmd tpm2_eventlog /sys/kernel/security/tpm0/binary_bios_measurements
+
+pcrs=(/sys/class/tpm/tpm0/pcr-sha*)
+as "At least one PCR bank is active" cukinia_test ${#pcrs[@]} -gt 0
+for pcr_bank in "${pcrs[@]}"; do
+    as "$(basename "$pcr_bank") 0..23 can be read" cukinia_cmd cat "$pcr_bank"/{0..23}
+done
+
+for pcr_bank in "${pcrs[@]}"; do
+    pcr17=$(cat "$pcr_bank/17" | grep -o . | uniq)
+    pcr18=$(cat "$pcr_bank/18" | grep -o . | uniq)
+    as "$(basename "$pcr_bank")/17 doesn't contain only zeroes or 'F's" \
+        cukinia_cmd [ "$pcr17" != "F" ] && [ "$pcr17" != "0" ]
+    as "$(basename "$pcr_bank")/18 doesn't contain only zeroes or 'F's" \
```

Comment:
For this "newer" TB release, can we still compare it against precalculated values, as we have in: https://gitlab.com/trenchboot1/3mdeb/testing-trenchboot @krystian-hebel 

I am not saying to include it here, I see it also rather as a test in OSFV.

Reply:
That wouldn't work for TXT, there are many different entries created by ACM that are hard to predict.

Reply:
Which means would work for AMD, and we want to include it in test scope?

Reply:
Maybe, but it would have to check on which CPU it runs, otherwise there would be false negative results.

Reply:
@macpijan removed in c154d49d2a9143b60d3d5822d38569af73024ca4